src/main/java/com/spotify/docker/client/DockerCertificates.java

```diff
@@ -149,8 +153,18 @@ public Builder clientCertPath(final Path clientCertPath) {
       return this;
     }
 
-    public DockerCertificates build() throws DockerCertificateException {
-      return new DockerCertificates(this);
+    public Optional<DockerCertificates> build() throws DockerCertificateException {
+      if (this.caCertPath == null || this.clientKeyPath == null || this.clientCertPath == null) {
+        log.debug("caCertPath, clientKeyPath or clientCertPath not specified, not using SSL");
+        return Optional.absent();
+      } else if (Files.exists(this.caCertPath) && Files.exists(this.clientKeyPath) &&
```

Comment:
I'd expect `Files.isReadable()` to throw an NPE when the supplied `Path` appeared to be `null`. I suppose checking for `null` here, combined with any exceptions later on (as pointed by @mattnworb), will be more clear to the user.